_search-plugins/search-pipelines/ml-inference-search-request.md

```diff
@@ -0,0 +1,532 @@
+
+```json
+"input_map": [  
+  {
+    "inputs": "query.term.label.value"
+  }
+]
+```
+
+**Step 2: Run the pipeline**
+
+Once you have created a search pipeline, you can run the same term query with the search pipeline:
+
+```json
+GET /my_index/_search?search_pipeline=my_pipeline_request_review
+{
+  "query": {
+    "term": {
+      "label": {
+        "value": "happy moments",
+        "boost": 1
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The query term value is rewritten based on the model's output. The model determines that the sentiment of the query term is positive, so the rewritten query looks like the following:
+
+```json
+{
+  "query": {
+    "term": {
+      "label": {
+        "value": "POSITIVE",
+        "boost": 1
+      }
+    }
+  }
+}
+```
+
+The response includes the document whose `label` field has the value `POSITIVE`:
+
+```json
+{
+  "took": 288,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.00009405752,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "3",
+        "_score": 0.00009405752,
+        "_source": {
+          "passage_text": "I am excited",
+          "passage_language": "en",
+          "label": "POSITIVE"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Example: Local model
+
+The following example configures an `ml_inference` processor with a local model to rewrite a term query into a k-NN query.
+
+**Step 1: Create a pipeline**
+
+The following example creates a search pipeline for the `huggingface/sentence-transformers/all-distilroberta-v1` local model. The model is a sentence transformer [pretrained model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers)
+hosted in your OpenSearch cluster.
+
+If you invoke the model using the Predict API, then the request looks like this:
+
+```json
+POST /_plugins/_ml/_predict/text_embedding/cleMb4kBJ1eYAeTMFFg4
+{
+  "text_docs": [
+    "today is sunny"
+  ],
+  "return_number": true,
+  "target_response": [
+    "sentence_embedding"
+  ]
+}
+```
+
+Using this schema, specify the `model_input` as follows:
+
+```json
+ "model_input": "{ \"text_docs\": ${input_map.text_docs}, \"return_number\": ${model_config.return_number}, \"target_response\": ${model_config.target_response} }"
+```
+
+In the `input_map`, map the `query.term.passage_embedding.value` query field to the `text_docs` field expected by the model:
+
+```json
+"input_map": [
+  {
+    "text_docs": "query.term.passage_embedding.value"
+  } 
+]
+```
+
+Because you specified the field to convert into embeddings as a JSON path, you need to set the `full_response_path` to `true` so that the full JSON document is parsed to obtain the input field:
```

Comment:
```suggestion
Because you specified the field to be converted into embeddings as a JSON path, you need to set the `full_response_path` to `true`. Then the full JSON document is parsed in order to obtain the input field:
```